frontend/src/entry/overview/EntryCardRolesChipSet.tsx

```diff
@@ -0,0 +1,22 @@
+import Dangerous from '@mui/icons-material/DangerousOutlined'
+import { Chip } from '@mui/material'
+import { CollaboratorEntry } from 'types/types'
+
+type EntryCardChipSetProps = {
+  entryCollaborator: CollaboratorEntry
+}
+
+export default function EntryCardRolesChipSet({ entryCollaborator }: EntryCardChipSetProps) {
```

Comment:
A few things here:
- Always use `sx` instead of `style` for MUI components. `sx` offers the same as `style`, but adds some MUI DX (developer experience) shortcuts.
- We should always be using colours defined in our theme, instead of defining colours inside components.
- The styling can be simplified a fair bit while still achieving the same effect.
- Having only one of the chips showing an icon looks a bit strange.
- We should `useMemo` the `.map` as mentioned in another comment (just a good habit to get into, though very minor in this specific case)

I've addressed all of the above here:
```
import { Chip } from '@mui/material'
import { useMemo } from 'react'
import { CollaboratorEntry } from 'types/types'

type EntryCardChipSetProps = {
  entryCollaborator: CollaboratorEntry
}

export default function EntryCardRolesChipSet({ entryCollaborator }: EntryCardChipSetProps) {
  const roleChips = useMemo(
    () => entryCollaborator.roles.map((role) => <Chip key={role} label={role} color='primary' sx={{ mr: 1 }} />),
    [entryCollaborator.roles],
  )

  return roleChips.length ? roleChips : <Chip label='No roles' color='warning' />
}

```